subprojects/internal-testing/src/main/groovy/org/gradle/test/fixtures/archive/JarTestFixture.groovy

```diff
@@ -21,20 +21,34 @@ import org.apache.commons.io.IOUtils
 import org.gradle.api.JavaVersion
 import org.gradle.internal.classanalysis.JavaClassUtil
 
+import javax.annotation.Nullable
 import java.util.jar.JarEntry
 import java.util.jar.JarFile
 import java.util.jar.Manifest
 
 class JarTestFixture extends ZipTestFixture {
     File file
+    private final @Nullable Manifest manifest
 
     /**
-     * Asserts that the Jar file is well-formed
+     * Asserts that the Jar file is well-formed.
      */
-     JarTestFixture(File file, String metadataCharset = 'UTF-8', String contentCharset = null) {
+    JarTestFixture(File file, String metadataCharset = 'UTF-8', String contentCharset = null) {
+        this(file, metadataCharset, contentCharset, true)
+    }
+
+    /**
+     * Creates the fixture.
+     */
+    JarTestFixture(File file, String metadataCharset, String contentCharset, boolean checkManifest) {
          super(file, metadataCharset, contentCharset)
          this.file = file
-         isManifestPresentAndFirstEntry()
+         if (checkManifest) {
+             isManifestPresentAndFirstEntry()
+             manifest = new Manifest(IOUtils.toInputStream(content('META-INF/MANIFEST.MF'), contentCharset as String))
```

Comment:
❓ Any particular reason not to load the manifest on demand?

Reply:
I intended to piggy-back on `isManifestPresentAndFirstEntry` initially, but discarded. Replaced with a lazy load (and made it possible to `manifest` even after creating the fixture with `checkManifest = false`).